### PR TITLE
Fix #225: Add tests in client/services/code_preprocessors

### LINKED ISSUE #225: add tests in client/services/code_preprocessors/
- [x] CodePreprocessorDispatcherService.js

- [x] PythonCodePreprocessorService.js

### PR DESCRIPTION
**Pull request by lumig242**
Fix #225 

Note: Seems that this is [line of code](https://github.com/google/tie/blob/master/client/services/code_preprocessors/PythonCodePreprocessorService.js#L152) will never be executed(which prevents 100% coverage of course).

There is nowhere to break the `while(true)` loop. I think it's safe to delete this line.

### CODE REVIEW

File: client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    it('should preprocess python source code', function(){
+      var codeSubmission = CodeSubmissionObjectFactory.create('a = 1\nb= 2');
+      var originProcessedCode = codeSubmission.getPreprocessedCode();
+      CodePreprocessorDispatcherService.preprocess(LANGUAGE_PYTHON,
+        codeSubmission, '', '', '', '', [], [], []);
+      expect(codeSubmission.getPreprocessedCode())
+        .not.toEqual(originProcessedCode);
+    });
+  });
+});
```
Comment (eyurko) [inline client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js]: newline at the end of files

File: client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    PythonCodePreprocessorService = $injector.get(
+      'PythonCodePreprocessorService');
+    LANGUAGE_PYTHON = $injector.get('LANGUAGE_PYTHON');
+    CodeSubmissionObjectFactory = $injector.get('CodeSubmissionObjectFactory');
+  }));
+
+  describe('preprocess', function() {
+    it('should throw error of Language not supported', function() {
+      var unsupportedLanguage = "someLanguage";
+      var failFunction = function() {
```
Comment (eyurko) [inline client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js]: failingFunction

File: client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    CodePreprocessorDispatcherService = $injector.get(
+      'CodePreprocessorDispatcherService');
+    PythonCodePreprocessorService = $injector.get(
+      'PythonCodePreprocessorService');
+    LANGUAGE_PYTHON = $injector.get('LANGUAGE_PYTHON');
+    CodeSubmissionObjectFactory = $injector.get('CodeSubmissionObjectFactory');
+  }));
+
+  describe('preprocess', function() {
+    it('should throw error of Language not supported', function() {
```
Comment (eyurko) [inline client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js]: should throw an error if the language passed in is not supported

File: client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js [hunk 4]
```diff
@@ -41,6 +41,18 @@ describe('PythonCodePreprocessorService', function() {
       ).toEqual("'stringify'");
     });
 
+    it('should correctly convert a json String with back slashes', function() {
+      expect(
+        PythonCodePreprocessorService._jsonVariableToPython('strin\\gify')
+      ).toEqual("'strin\\\\gify'");
+    });
+
+    it('should correctly convert a json number to a Python number', function() {
```
Comment (eyurko) [inline client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js]: integer, not number

File: client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js [hunk 5]
```diff
@@ -41,6 +41,18 @@ describe('PythonCodePreprocessorService', function() {
       ).toEqual("'stringify'");
     });
 
+    it('should correctly convert a json String with back slashes', function() {
```
Comment (eyurko) [inline client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js]: a JSON string with escaped characters

File: client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js [hunk 6]
```diff
@@ -41,6 +41,18 @@ describe('PythonCodePreprocessorService', function() {
...
+
+    it('should correctly convert a json number to a Python number', function() {
+      expect(
+        PythonCodePreprocessorService._jsonVariableToPython(12)
+      ).toEqual(12);
+    });
+
     it(
       [
         'should correctly convert a json Array to a string version ',
```
Comment (eyurko) [inline client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js]: while you're in here, just change json everywhere to JSON -- it's an acronym

File: client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js [hunk 7]
```diff
@@ -69,6 +81,15 @@ describe('PythonCodePreprocessorService', function() {
           [[true, true], [false, false], [true, false]])
       ).toEqual("[[True, True], [False, False], [True, False]]");
     });
+
+    it('should throw error for unsupported inputs', function() {
```
Comment (eyurko) [inline client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js]: should throw an error if the input type is unsupported

**Comment by eyurko** [comment]
Feel free to delete that line as well.

File: client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js (hunk 1 above)
Comment (lumig242) [inline client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js]: New line added

**Comment by lumig242** [comment]
@eyurko Please take another look when you are free:)

File: client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js [hunk 8]
```diff
@@ -35,23 +35,37 @@ describe('PythonCodePreprocessorService', function() {
   }));
 
   describe('_jsonVariableToPython', function() {
-    it('should correctly convert a json String to a Python string', function() {
+    it('should correctly convert a JSON String to a Python string', function() {
```
Comment (eyurko) [inline client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js]: nit: string and array don't need to be capitalized.

**Pull request merged**